extension/entitystore/extension_test.go

```diff
@@ -332,12 +344,115 @@ func TestEntityStore_shouldReturnRID(t *testing.T) {
+			assert.Equalf(t, tt.nextRefreshTime, e.nextRefreshTime, "nextRefreshTime")
+			assert.Equal(t, 1, e.stsClient.(*mockSTSClient).callCount)
 		})
 	}
 }
 
+func TestNumberOfStsCalls(t *testing.T) {
+	type fields struct {
+		metadataprovider ec2metadataprovider.MetadataProvider
+		stsClient        stsiface.STSAPI
+		nativeCredential client.ConfigProvider
+	}
+	instanceId := "i-abcd1234"
+	accountId := "123456789012"
+	glob := LogFileGlob("glob")
+	group := LogGroupName("group")
+	serviceAttr := ServiceAttribute{
+		ServiceName:       "test-service",
+		ServiceNameSource: ServiceNameSourceUserConfiguration,
+		Environment:       "test-environment",
+	}
+	sp := new(mockServiceProvider)
+	sp.On("logFileServiceAttribute", glob, group).Return(serviceAttr)
+	tests := []struct {
+		name                  string
+		fields                fields
+		numberOfFunctionCalls int
+		wantNumberOfStsCalls  int
+		mockTimeSkipInterval  int64
+	}{
+		{
+			name: "1_function_call",
+			fields: fields{
+				metadataprovider: mockMetadataProviderWithAccountId(accountId),
+				stsClient:        &mockSTSClient{accountId: accountId},
+				nativeCredential: &session.Session{},
+			},
+			numberOfFunctionCalls: 1,
+			wantNumberOfStsCalls:  1,
+		},
+		{
+			name: "50_function_calls_immediate",
+			fields: fields{
+				metadataprovider: mockMetadataProviderWithAccountId(accountId),
+				stsClient:        &mockSTSClient{accountId: accountId},
+				nativeCredential: &session.Session{},
+			},
+			numberOfFunctionCalls: 50,
+			wantNumberOfStsCalls:  1,
+		},
+		{
+			name: "50_function_calls_after_successful_interval",
+			fields: fields{
+				metadataprovider: mockMetadataProviderWithAccountId(accountId),
+				stsClient:        &mockSTSClient{accountId: accountId},
+				nativeCredential: &session.Session{},
+			},
+			numberOfFunctionCalls: 50,
+			wantNumberOfStsCalls:  50,
+			mockTimeSkipInterval:  int64(successfulRefresh),
+		},
+		{
+			name: "50_function_calls_after_failure_interval",
+			fields: fields{
+				metadataprovider: mockMetadataProviderWithAccountId("210987654321"),
+				stsClient:        &mockSTSClient{accountId: accountId},
+				nativeCredential: &session.Session{},
+			},
+			numberOfFunctionCalls: 50,
+			wantNumberOfStsCalls:  50,
+			mockTimeSkipInterval:  int64(failureRefresh),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var currentTime = time.Now()
+			e := EntityStore{
+				mode:             config.ModeEC2,
+				ec2Info:          ec2Info{InstanceID: instanceId},
+				serviceprovider:  sp,
+				metadataprovider: tt.fields.metadataprovider,
+				stsClient:        tt.fields.stsClient,
+				nativeCredential: tt.fields.nativeCredential,
+				currentTime: func() time.Time {
+					currentTime = currentTime.Add(time.Duration(tt.mockTimeSkipInterval) + 1)
+					return currentTime
+				},
+			}
+			for i := 0; i < tt.numberOfFunctionCalls; i++ {
+				e.CreateLogFileEntity(glob, group)
```

Comment:
This is the function called by the logs agent when making the PLE call, so thought it was best to test by calling this function.